libs/lz4/src/main/java/org/elasticsearch/lz4/ESLZ4Decompressor.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+/*
+ * Copyright 2020 Adrien Grand and the lz4-java contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.elasticsearch.lz4;
+
+import net.jpountz.lz4.LZ4Exception;
+import net.jpountz.lz4.LZ4FastDecompressor;
+
+import java.nio.ByteBuffer;
+
+/**
+ * This file is forked from https://github.com/lz4/lz4-java. In particular, it forks the following file
+ * net.jpountz.lz4.LZ4JavaSafeFastDecompressor.
+ *
+ * It modifies the original implementation to use custom LZ4SafeUtils and SafeUtils implementations which
+ * include performance improvements.
+ */
+public class ESLZ4Decompressor extends LZ4FastDecompressor {
+    public static final LZ4FastDecompressor INSTANCE = new ESLZ4Decompressor();
+
+    ESLZ4Decompressor() {
+    }
+
+    public int decompress(byte[] src, int srcOff, byte[] dest, int destOff, int destLen) {
+        SafeUtils.checkRange(src, srcOff);
+        SafeUtils.checkRange(dest, destOff, destLen);
+        if (destLen == 0) {
+            if (SafeUtils.readByte(src, srcOff) != 0) {
+                throw new LZ4Exception("Malformed input at " + srcOff);
+            } else {
+                return 1;
+            }
+        } else {
+            int destEnd = destOff + destLen;
+            int sOff = srcOff;
+            int dOff = destOff;
+
+            while(true) {
+                int token = SafeUtils.readByte(src, sOff) & 255;
+                ++sOff;
+                int literalLen = token >>> 4;
+                if (literalLen == 15) {
+                    byte len;
+                    for(boolean var11 = true; (len = SafeUtils.readByte(src, sOff++)) == -1; literalLen += 255) {
```

Comment:
`var11` seems to be some  decompiling artifact? It doesn't seem to be used anywhere.